internal/promote/storage.go

```diff
@@ -98,37 +98,72 @@ func (prs PackageVersions) Strings() []string {
 
 // CloneRepository method clones the repository and changes branch to stage.
 func CloneRepository(user, stage string) (*git.Repository, error) {
-	r, err := git.Clone(memory.NewStorage(), memfs.New(), &git.CloneOptions{
-		URL:           fmt.Sprintf(repositoryURL, "elastic"),
-		RemoteName:    remoteName,
-		ReferenceName: plumbing.NewBranchReferenceName(stage),
+	// Initialize repository
+	r, err := git.Init(memory.NewStorage(), memfs.New())
+	if err != nil {
+		return nil, errors.Wrap(err, "initializing repository")
+	}
+
+	// Add remotes
+	userRepositoryURL := fmt.Sprintf(repositoryURL, user)
+	userRemote, err := r.CreateRemote(&config.RemoteConfig{
+		Name: user,
+		URLs: []string{
+			userRepositoryURL,
+		},
 	})
 	if err != nil {
-		return nil, errors.Wrap(err, "cloning package-storage repository failed")
+		return nil, errors.Wrap(err, "creating user remote failed")
+	}
+	remote, err := r.CreateRemote(&config.RemoteConfig{
+		Name: remoteName,
+		URLs: []string{
+			fmt.Sprintf(repositoryURL, "elastic"),
+		},
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "creating remote failed")
 	}
 
-	err = r.Fetch(&git.FetchOptions{
-		RemoteName: remoteName,
+	// Check if user remote exists
```

Comment:
Along the same lines as this, do you think it would be good to check if the upstream remote exists before the fetch and checkout step below?

Reply:
Upstream is going to be indirectly checked when doing the fetch, I think this would be enough.